tests/unit/llms/test_llm_openai.py

```diff
@@ -54,3 +58,151 @@ def test_openai_llm_call_success(self, openai_llm, mocker):
         llm_input = [Message(role="user", content="test")]
         output = openai_llm(llm_input)
         assert output == "test"
+
+    def test_get_schema_openai_with_valid_callable(self):
+        def sample_function(param1: int, param2: str = "default") -> str:
+            """Sample function for testing."""
+            return f"param1: {param1}, param2: {param2}"
+
+        expected_schema = {
+            "type": "function",
+            "function": {
+                "name": "sample_function",
+                "description": "Sample function for testing.",
+                "parameters": {
+                    "type": "object",
+                    "properties": {
+                        "param1": {
+                            "type": "number",
+                            "description": "No description available.",
+                        },
+                        "param2": {
+                            "type": "string",
+                            "description": "No description available.",
+                        },
+                    },
+                    "required": ["param1"],
+                },
+            },
+        }
+        schema = get_schema_openai(sample_function)
+        assert schema == expected_schema, "Schema did not match expected output."
+
+    def test_get_schema_openai_with_non_callable(self):
+        non_callable = "I am not a function"
+        with pytest.raises(ValueError):
+            get_schema_openai(non_callable)
+
+    def test_openai_llm_call_with_function_schema(self, openai_llm, mocker):
+        mock_completion = mocker.MagicMock()
+        mock_completion.choices[0].message.tool_calls = [
+            mocker.MagicMock(function=mocker.MagicMock(arguments="result"))
+        ]
+        mocker.patch.object(
+            openai_llm.client.chat.completions, "create", return_value=mock_completion
+        )
+        llm_input = [Message(role="user", content="test")]
+        function_schemas = [{"type": "function", "name": "sample_function"}]
+        output = openai_llm(llm_input, function_schemas)
+        assert (
+            output == "result"
+        ), "Output did not match expected result with function schema"
+
+    def test_openai_llm_call_with_invalid_tool_calls(self, openai_llm, mocker):
```

Comment:
We should be confirming that the way we do and call single schemas, multiple schemas, OpenAI, Cohere, or Local LLMs is _exactly_ the same for the users, we need standard input and output for all otherwise it becomes a mess for users to understand when they should do X or Y or Z

Reply:
Okay I guess the change to this test will follow form the above changes.

Reply:
sounds good